src/HW_models/NRF_DPPI.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "NRF_DPPI.h"
+#include <stdlib.h>
+#include "bs_tracing.h"
+#include <string.h>
+
+/*
+ * DPPI — Distributed programmable peripheral interconnect
+ * https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Fkeyfeatures_html5.html&cp=4_0_0
+ *
+ */
+
+NRF_DPPIC_Type NRF_DPPI_regs; ///< The DPPI registers
+
+/**
+ * DPPI module own TASKs handlers
+ */
+void nrf_dppi_TASK_CHG_ENDIS( int groupnbr, bool enable /*false=disable task*/ )
+{
+  if ( enable )
+  {
+    bs_trace_raw_time(9, "dppi: Channel group %i enabled\n", groupnbr);
+    NRF_DPPI_regs.CHEN |= NRF_DPPI_regs.CHG[groupnbr];
+  } else
+  {
+    bs_trace_raw_time(9 ,"dppi: Channel group %i disable\n", groupnbr);
+    NRF_DPPI_regs.CHEN &= ~NRF_DPPI_regs.CHG[groupnbr];
+  }
+  //Note of the author: From the spec I cannot guess if these tasks will affect
+  //CHEN or a separate hidden register
+}
+
+void nrf_dppi_TASK_CHG0_EN() { nrf_dppi_TASK_CHG_ENDIS(0,true); }
+void nrf_dppi_TASK_CHG1_EN() { nrf_dppi_TASK_CHG_ENDIS(1,true); }
+void nrf_dppi_TASK_CHG2_EN() { nrf_dppi_TASK_CHG_ENDIS(2,true); }
+void nrf_dppi_TASK_CHG3_EN() { nrf_dppi_TASK_CHG_ENDIS(3,true); }
+void nrf_dppi_TASK_CHG4_EN() { nrf_dppi_TASK_CHG_ENDIS(4,true); }
+void nrf_dppi_TASK_CHG5_EN() { nrf_dppi_TASK_CHG_ENDIS(5,true); }
+void nrf_dppi_TASK_CHG0_DIS(){ nrf_dppi_TASK_CHG_ENDIS(0,false); }
+void nrf_dppi_TASK_CHG1_DIS(){ nrf_dppi_TASK_CHG_ENDIS(1,false); }
+void nrf_dppi_TASK_CHG2_DIS(){ nrf_dppi_TASK_CHG_ENDIS(2,false); }
+void nrf_dppi_TASK_CHG3_DIS(){ nrf_dppi_TASK_CHG_ENDIS(3,false); }
+void nrf_dppi_TASK_CHG4_DIS(){ nrf_dppi_TASK_CHG_ENDIS(4,false); }
+void nrf_dppi_TASK_CHG5_DIS(){ nrf_dppi_TASK_CHG_ENDIS(5,false); }
+
+/*Forward declaration*/
+struct subscr_list_elem;
+
+//Element of linked list of subscribers
+typedef struct subscr_list_elem
+{
+  dest_f_t task_function;
+  struct subscr_list_elem * p_next;
+} subscr_list_elem_t;
+
+//Table with TASKs subscribing to all channels
+static subscr_list_elem_t* dppi_subscriber_list[NUMBER_DPPI_CHANNELS];
+
+//Function for checking if an element is in the subscriber linked list
+static bool subscriber_list_check(dest_f_t task_function, subscr_list_elem_t ** pp_head)
```

Comment:
No need to use pointer to pointer in this function. You may use
`subscriber_in_list(dest_f_t task_function, subscr_list_elem_t * p_head)`?